mifosng-android/src/release/java/com/mifos/api/local/MifosSQLCipherHelper.java

```diff
@@ -0,0 +1,18 @@
+package com.mifos.api.local;
+
+import com.raizlabs.android.dbflow.config.DatabaseDefinition;
+import com.raizlabs.android.dbflow.structure.database.DatabaseHelperListener;
+import com.raizlabs.dbflow.android.sqlcipher.SQLCipherOpenHelper;
+
+public class MifosSQLCipherHelper extends SQLCipherOpenHelper {
+
+    public MifosSQLCipherHelper(DatabaseDefinition databaseDefinition,
+                                DatabaseHelperListener listener) {
+        super(databaseDefinition, listener);
+    }
+
+    @Override
+    protected String getCipherSecret() {
+        return "dbflow-rules";
```

Comment:
@satyan Suggestion

One of the options could be to use user's password entry to encrypt the key on the client. So you could only decrypt with the help of user.

It adds a layer of entropy to the system. However, could mean that we ask the user for password multiple times.. 

Android also supports a hardware backed keychain support. You could potentially explore that as well to store the key.

Having a server backed key is a good option too. Except, in offline case, it'd be difficult to obtain. Do not store in shared preferences, as it's stored in a file on the device in clear. Easily obtained.